src/ResourceManager/DataMigration/Commands.DataMigration/Cmdlets/NewDataMigrationTask.cs

```diff
@@ -176,12 +216,29 @@ public override void ExecuteCmdlet()
                     ProjectTask response = null;
                     try
                     {
+                        ProjectTaskProperties properties = taskCmdlet.ProcessTaskCmdlet();
+
+                        var utcStartedOn = System.DateTime.UtcNow;
+                        // need swagger of -pr line 64 of tasks.json add.
+                        // uncomment the following line once we get new sdk.
+                        // give all tasks a start time, so that portal can calculate how long it is running.
+                        // properties.ClientData.Add("startedOn", utcStartedOn.ToString("o")); 
+
                         ProjectTask taskInput = new ProjectTask()
                         {
-                            Properties = taskCmdlet.ProcessTaskCmdlet()
+                            Properties = properties
                         };
 
                         response = DataMigrationClient.Tasks.CreateOrUpdate(taskInput, ResourceGroupName, ServiceName, ProjectName, Name);
+
+                        // wait for the task to finish: not queued or running state:
+                        while (this.Wait.IsPresent && response !=null && response.Properties != null &&
+                            ( response.Properties.State == "Queued" || response.Properties.State == "Running" ) )
+                        {
+                            System.Threading.Thread.Sleep(5000);
```

Comment:
@haitsongmsft nit: I'd replace `5000` with `TimeSpan.FromSeconds(5))`, as well as replacing the `5` with a variable at the top of this file that defines the value.

Reply:
updated